### ISSUE TITLE
Bug: [no-base-to-string] should check if superclass is ignored

### ISSUE THREAD

**Issue by tyilo** [typescript-eslint/typescript-eslint#11520]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=JYOwLgpgTgZghgYwgAgLIE8CiUoHsrIQAekIAJgM7LZ4EDeyAvgFDNkQIA2cUKCuICmEI58ALjRZRUANzNotAHRhcAZTBRQAcwAUAShlA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oCMBDZRLXxdk%2BKkwDm6MAG1w2HImjQO0SABp5CrAuyRKEztEQATAColEAOV4BbFNLm7dkAKLLVkLboC63iAC%2B3n6BIAFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
interface MyError extends Error { }

declare const error: MyError;
error.toString();
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-base-to-string": [
      "error",
      {
        "ignoredTypeNames": [
          "Error"
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

There should be no lints.

### Actual Result

```
'error' will use Object's default stringification format ('[object Object]') when stringified. 4:1 - 4:6
```

### Additional Info

There is some discussion in https://github.com/typescript-eslint/typescript-eslint/issues/11420

**Comment by JoshuaKGoldberg** [comment]
👍 Agreed, an instance of a sub-class _is_ an instance of the base class.
